### PR TITLE
Test coverage by rspec-puppet and puppetlabs_spec_helper integration

### PR DESCRIPTION
**Pull request by dkerwin**
Hi guys,

i added some tests to your cassandra module. PR includes:
- Switch to puppetlabs_spec_helper
- 91 rspec tests to cover all classes and parameter validation
- Added missing dependency to puppetlabs/apt
- Updated Gemfile to track test depencies

Cheers :beers: 

Daniel

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,9 +1,4 @@
-require 'rake'
-require 'rspec/core/rake_task'
-require 'puppet-lint/tasks/puppet-lint'
```
Comment (joschi) [inline Rakefile:3]: Did you remove puppet-lint by intention? It currently isn't automatically executed but I wanted to add a lint check to the CI job.

File: manifests/init.pp [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
+    $concurrent_reads           = $cassandra::params::concurrent_reads,
+    $concurrent_writes          = $cassandra::params::concurrent_writes,
+    $incremental_backups        = $cassandra::params::incremental_backups,
     $snapshot_before_compaction = $cassandra::params::snapshot_before_compaction,
-    $auto_snapshot = $cassandra::params::auto_snapshot,
-    $multithreaded_compaction = $cassandra::params::multithreaded_compaction,
-    $endpoint_snitch = $cassandra::params::endpoint_snitch,
+    $auto_snapshot              = $cassandra::params::auto_snapshot,
+    $multithreaded_compaction   = $cassandra::params::multithreaded_compaction,
+    $endpoint_snitch            = $cassandra::params::endpoint_snitch,
```
Comment (joschi) [inline manifests/init.pp]: The Puppet 2.6 parser has problems with trailing commas in parameter lists. If you remove it, the Travis CI jobs with Puppet 2.6 will hopefully work again.

**Comment by joschi** [comment]
Yay, thanks for effort! :+1:

File: Rakefile (hunk 1 above)
Comment (dkerwin) [inline Rakefile:3]: It's automatically enabled in puppetlabs_spec_helper. Here are the builtin rake tasks:

```
⌘ > rake -T
rake build            # Build puppet module package
rake clean            # Clean a built module package
rake coverage         # Generate code coverage information
rake help             # Display the list of available rake tasks
rake lint             # Check puppet manifests with puppet-lint
rake spec             # Run spec tests in a clean fixtures directory
rake spec_clean       # Clean up the fixtures directory
rake spec_prep        # Create the fixtures directory
rake spec_standalone  # Run spec tests on an existing fixtures directory
```

Comment (joschi) [inline Rakefile:3]: Ah, thanks for explaining. I didn't know that.

File: manifests/init.pp (hunk 2 above)
Comment (dkerwin) [inline manifests/init.pp]: Thanks. Missed that one. Results in even stranger message :anger: 

`Syntax error at '['; expected ']' at /Users/dkerwi/dev/puppet-cassandra/spec/fixtures/modules/cassandra/manifests/init.pp:46 on node`

Comment (joschi) [inline manifests/init.pp]: That's caused by a too recent version of stdlib. The `.travis.yml` and `.fixtures.yml` don't specify any branch or tag but always fetch the `master` branch of puppetlabs/puppetlabs-stdlib. And only stdlib <= 2.0 is compatible with Puppet 2.6, see https://github.com/puppetlabs/puppetlabs-stdlib#compatibility

**Comment by dkerwin** [comment]
Finally. I changed the valid_re matchers as arrays cause trouble in 2.6. The single regex might by faster as well. Also excluded 1.9.3/2.6.14 from build matrix as not supported by puppetlabs (http://docs.puppetlabs.com/guides/platforms.html#ruby-versions).

Cheers!

**Comment by joschi** [comment]
Woohoo!

Thank you very much for the patch and cheers!  :cocktail:

**Pull request merged**